source/interprocedural_analyses/taint/taintConfiguration.ml

```diff
@@ -687,11 +709,41 @@ module SharedMemory = struct
       None
 end
 
+module TaintConfigLocation = struct
+  type t = {
+    path: PyrePath.t option;
+    location: JsonParsing.Location.t option;
+  }
+  [@@deriving equal, compare]
+
+  let pp_location { path; location } =
+    let file_location =
+      match path with
+      | Some pyre_path -> PyrePath.show pyre_path
+      | None -> "possibly the internal rules list"
+    in
+    let location_in_file =
+      match location with
+      | Some { JsonParsing.Location.start_line; end_line; start_column; end_column } ->
+          Format.sprintf
+            "from line `%d` column `%d` to line `%d` column `%d`"
+            start_line
+            end_line
+            start_column
+            end_column
+      | None -> "at unknown location"
+    in
+    Format.sprintf "`%s` %s" file_location location_in_file
+end
+
 module Error = struct
   type kind =
     | FileNotFound
     | FileRead
-    | InvalidJson of string
+    | InvalidJson of {
+        message: string;
+        location: JsonParsing.Location.t;
```

Comment:
ditto previous PR, could we store the location directly in `type t`?

Reply:
Yes!